arangod/Aql/ExecutionNode.h

```diff
@@ -487,6 +486,15 @@ class ExecutionNode {
   auto getRegsToKeepStack() const -> RegIdSetStack;
 
  protected:
+  /// @brief serialize this ExecutionNode to VelocyPack
+  void toVelocyPack(arangodb::velocypack::Builder&, unsigned flags) const;
```

Comment:
For Logging / Testing / Debugging Purposes we could even make this Method Public, as it will Serialize a single Node standalone (it cannot make sense if its dependencies but everything else should be self-contained)

Reply:
Yes, we might have ambiguities with `CollectionAccessingNode::toVelocyPack`, `DocumentProducingNode::toVelocyPack` and possibly others, but these can be sorted out. This should also simplify some unit tests. Will do, and also add a dedicated test for `allToVelocyPack`.